src/util/flattenReactChildren.ts

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+type ReactChildArray = ReturnType<typeof React.Children.toArray>;
+
+export function flattenReactChildren(
+  children: React.ReactNode,
+): ReactChildArray {
+  const childrenArray = React.Children.toArray(children);
+  return childrenArray.reduce((flatChildren: ReactChildArray, child) => {
+    if ((child as React.ReactElement<any>).type === React.Fragment) {
+      return flatChildren.concat(
+        flattenReactChildren((child as React.ReactElement<any>).props.children),
+      );
+    }
+    flatChildren.push(child);
+    return flatChildren;
```

Comment:
Could you do
```suggestion
    return [...flatChildren, child];
```
?

Reply:
Actually I think it's better to keep it this way to prevent multiple iterations of destructuring `flatChildren`